src/Microsoft.AspNetCore.Blazor.Razor.Extensions/BlazorRuntimeNodeWriter.cs

```diff
@@ -304,6 +306,26 @@ public override void WriteHtmlContent(CodeRenderingContext context, HtmlContentI
             }
         }
 
+        private void RejectDisallowedHtmlTags(IntermediateNode node, HtmlTagToken tagToken)
+        {
+            // Disallow <script> in components as per #552
+            // Case-sensitive comparison is fine because AngleSharp always lowercases tag names
+            if (tagToken.Name.Equals("script", StringComparison.Ordinal))
+            {
+                const string suppressErrorAttributeName = "suppress-error";
+                if (string.Equals(tagToken.GetAttribute(suppressErrorAttributeName), "BL9992", StringComparison.Ordinal))
+                {
+                    tagToken.Attributes.RemoveAll(kvp => kvp.Key.Equals(suppressErrorAttributeName, StringComparison.Ordinal));
+                }
+                else
+                {
+                    var adjustedSpan = CalculateSourcePosition(node.Source, tagToken.Position);
+                    var diagnostic = BlazorDiagnosticFactory.Create_DisallowedScriptTag(adjustedSpan);
+                    throw new RazorCompilerException(diagnostic);
```

Comment:
I think this is fine for now, but keep in mind that any time you throw and catch in the compiler, that means that someone hitting the error will lose all intellisense and colorization in the rest of the file after this point. If at all possible, we should just skip the offending tag and its contents.

Reply:
Is that the case for code in the runtime node writer as well as the design-time one?

Reply:
The problem with the runtime writer is that you won't run through the rest of the file, so if you throw here it will suppress any other errors in the file.

I don't have a problem with doing this as a temporary fix. I have a todo to remove all of these and the exception.